core/src/main/java/io/aiven/klaw/service/MailUtils.java

```diff
@@ -355,6 +355,35 @@ private void sendMail(
         });
   }
 
+  private void sendPwdResetMail(
+      String username,
+      HandleDbRequests dbHandle,
+      String formattedStr,
+      String subject,
+      boolean registrationRequest,
+      String otherMailId,
+      int tenantId,
+      String loginUrl) {
+
+    CompletableFuture.runAsync(
+        () -> {
+          String emailId;
+
+          try {
+            emailId = dbHandle.getUsersInfo(username).getMailid();
```

Comment:
How about using method selectAllCachedUserInfo fromManageDatabase through commonUtils class, and stream/filter?